lib/Elevate/Leapp.pm

```diff
@@ -221,6 +223,41 @@ sub extract_error_block_from_output ( $self, $text_ar ) {
     return $error_block;
 }
 
+sub check_upgrade_log_for_failures ($self) {
+    my $upgrade_log = LEAPP_UPGRADE_LOG;
+    my $cont_file   = LEAPP_FAIL_CONT_FILE;
+
+    if ( !-e $upgrade_log ) {
+        ERROR("No LEAPP upgrade log detected at [$upgrade_log]");
+
+        return 1;
+    }
+
+    my $grep_firstboot = $self->cpev->ssystem_capture_output( '/usr/bin/grep', '"Starting stage After of phase FirstBoot"', $upgrade_log );
+
+    if ( !scalar @{ $grep_firstboot->{'stdout'} } ) {
+        if ( -e $cont_file ) {
+            INFO("LEAPP does not appear to have completed successfully, but continuing anyway because [$cont_file] exists");
+
+            return 0;
+        }
+        else {
+            my $msg = sprintf("The command 'leapp upgrade' did not complete successfully. Please investigate and resolve the issue.\n");
+            $msg .= sprintf("Once resolved, you can continue the upgrade by running the following commands:\n");
+            $msg .= sprintf( "\t%s\n", "touch $cont_file" );
+            $msg .= sprintf( "\t%s\n", '/scripts/elevate-cpanel --continue' );
+            $msg .= sprintf("The following log files may help in your investigation.\n");
+            $msg .= sprintf( "\t%s\n", $upgrade_log );
+            $msg .= sprintf( "\t%s\n", LEAPP_REPORT_TXT );
```

Comment:
CONSIDER:  use a preset number of spaces instead tabs.  In my experience, using tabs in production code leads to parsing problems long term due to the way different OSs handle tabs.

Reply:
it also confuses terminals and systemd sometimes.